fabric-mining-levels-v0/src/main/java/net/fabricmc/fabric/mixin/tools/MixinItemStack.java

```diff
@@ -46,7 +47,10 @@ public void getBlockBreakingSpeed(BlockState state, CallbackInfoReturnable<Float
 		if (this.getItem() instanceof MiningToolItemAccessor) {
 			TriState triState = ToolManager.handleIsEffectiveOn((ItemStack) (Object) this, state);
 			if (triState != TriState.DEFAULT) {
-				info.setReturnValue(triState.get() ? ((MiningToolItemAccessor) this.getItem()).getMiningSpeed() : 1.0F);
+				float miningSpeed;
+				if (this.getItem() instanceof DynamicTool) miningSpeed = ((DynamicTool) this.getItem()).getMiningSpeed((ItemStack)(Object) this);
```

Comment:
I would suggest adding braces around this if statement, I dont know if its just me but its kinda hard to follow, there is no reason to compress it up like this.